### PR TITLE
WIP Support MacOS build and test

### PR DESCRIPTION
**Pull request by timkpaine**
Fixes https://github.com/chipsalliance/synlig/issues/1980

Will not work yet until a few other things are finished, so please do not approve/run the CI (I'm testing on my fork)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -159,6 +160,47 @@ jobs:
...
+        with:
+          path: |
+              ~/Library/Caches/Homebrew/boost--*
+              ~/Library/Caches/Homebrew/downloads/*--boost-*
+          key: brew-${{ matrix.os }}
+          restore-keys: brew-
+
+      - name: Install Mac Dependencies
+        run: |
+          brew install uhdm surelog yosys
```
Comment (kgugala) [inline .github/workflows/main.yml:190]: I suppose we should build everything here (not try to install those packages with `brew`), so we're sure we test against the exact version of the deps pinned with submodules

**Review by kgugala — CHANGES_REQUESTED** [review]

**Comment by kgugala** [comment]
@timkpaine can you rebase this PR (so we have changes from #2073 included)

File: .github/workflows/main.yml (hunk 1 above)
Comment (timkpaine) [reply .github/workflows/main.yml:190]: Sure, though of course surelog and uhdm are available via homebrew, and soon sunlit will be as well.

so long as the dependencies are pinned to release versions and not random git hashes, it should be easy to distribute via homebrew and other registries.

Comment (pgielda) [reply .github/workflows/main.yml:190]: Actually I think we should have both tests - one without and one with brew.

**Pull request closed without merging**